### PR TITLE
releases: add 3.1.0-rc release notes

### PR DESCRIPTION
**Pull request by anotherrachel**
### What is changed, added or deleted? (Required)
Add 3.1.0-rc release notes

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version, including v4.0 changes for now)
- [x] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

File: releases/3.1.0-rc.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+发版日期：2020 年 4 月 2 日
+
+TiDB 版本：3.1.0-rc
+
+TiDB Ansible 版本：3.1.0-rc
+
+## 新功能
+
++ TiDB
+    - 采用的二分搜索实现分区裁剪，以来提升性能[#15678](https://github.com/pingcap/tidb/pull/15678)
```
Comment (TomShawn) [inline releases/3.1.0-rc.md:17]:
```suggestion
    - 采用的二分搜索实现分区裁剪，以来提升性能 [#15678](https://github.com/pingcap/tidb/pull/15678)
```

File: releases/3.1.0-rc.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    - 支持 TiDB 在每次新建连接时动态更新 TLS 证书，实现不重启更新过期客户端证书 [#15163](https://github.com/pingcap/tidb/pull/15163)
+    - 通过更新 PD Client 支持每次新建连接是读取加载最新的证书 [#15425](https://github.com/pingcap/tidb/pull/15425)
+    - 如果配置了 `Cluster-SSL-*` 强制让 TiDB-PD 和 TiDB-TiDB 使用配置的证书进行 HTTPS 协议传输[#15430](https://github.com/pingcap/tidb/pull/15430)
+    - 新增和 MySQL 兼容的 `--require-secure-transport` 启动项，配置时强制客户端使用 TLS [#15442](https://github.com/pingcap/tidb/pull/15442)
+    - 添加 `cluster-verify-cn` 配置，只有拥有特定 CN 属性值证书的访问者才能访问 TiDB Status Port 或建立 gRPC 连接 [#15137](https://github.com/pingcap/tidb/pull/15137)
+
++ TiKV
+    - 支持通过 Raw KV API 备份数据 [#7051](https://github.com/tikv/tikv/pull/7051)
+    - 状态服务支持 TLS [#7142](https://github.com/tikv/tikv/pull/7142)
+    - KV server 支持 TLS[#7305](https://github.com/tikv/tikv/pull/7305)
```
Comment (TomShawn) [inline releases/3.1.0-rc.md:33]:
```suggestion
    - KV server 支持 TLS [#7305](https://github.com/tikv/tikv/pull/7305)
```

File: releases/3.1.0-rc.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
++ TiKV
+    - 修复在关闭一致性检查参数时，事务中插入已存在的 Key 且立马删除导致冲突检测失效或数据索引不一致的问题 [#7112](https://github.com/tikv/tikv/pull/7112)
+    - 修复 `TopN` 比较无符号整型时计算错误的问题 [#7199](https://github.com/tikv/tikv/pull/7199)
+    - Raftstore 引入流控机制，解决没有流控可能导致追日志太慢可能导致集群卡住，以及事务大小太大会导致 TiKV 间连接频繁重连的问题 [#7087](https://github.com/tikv/tikv/pull/7087) [#7078](https://github.com/tikv/tikv/pull/7078)
+    - 修复发送到 replicas 的读请求可能被永久卡住的问题 [#6543](https://github.com/tikv/tikv/pull/6543)
+    - 修复 replica read 会被 apply snapshot 阻塞的问题 [#7249](https://github.com/tikv/tikv/pull/7249)
+    - 修复 read index 在 transfer leader 情况下可能导致 panic 的问题 [#7240](https://github.com/tikv/tikv/pull/7240)
+    - 修复备份到 S3 时所有 SST 文件填充为零的问题 [#6967](https://github.com/tikv/tikv/pull/6967)
+    - 修复备份时未记录 SST 文件大小的导致恢复后有很多空 Region 的问题 [#6983](https://github.com/tikv/tikv/pull/6983)
+    - 优化持有锁的时间， 的性能 [#7202](https://github.com/tikv/tikv/pull/7202)
```
Comment (TomShawn) [inline releases/3.1.0-rc.md:79]: @scsldb PTAL
```suggestion
    - 优化持有锁的时间以提升备份性能 [#7202](https://github.com/tikv/tikv/pull/7202)
```

File: releases/3.1.0-rc.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+TiDB 版本：3.1.0-rc
+
+TiDB Ansible 版本：3.1.0-rc
+
+## 新功能
+
++ TiDB
+    - 采用的二分搜索实现分区裁剪，以来提升性能[#15678](https://github.com/pingcap/tidb/pull/15678)
+    - 支持 `RECOVER` 语法恢复被 truncate table 的数据 [#15460](https://github.com/pingcap/tidb/pull/15460)
```
Comment (TomShawn) [inline releases/3.1.0-rc.md:18]:
```suggestion
    - 支持 `RECOVER` 语法恢复被 `truncate table` 删除的数据 [#15460](https://github.com/pingcap/tidb/pull/15460)
```

File: releases/3.1.0-rc.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    - 支持重用语句重试中已分配的 `AUTO_RANDOM` ID [#15393](https://github.com/pingcap/tidb/pull/15393)
+    - 支持 `recover table` 恢复 `AUTO_RANDOM` ID 分配器的状态 [#15393](https://github.com/pingcap/tidb/pull/15393)
+    - 支持 `YEAR`, `MONTH`, `TO_DAY` 函数作为 Hash partition table 的分区 key [#15619](https://github.com/pingcap/tidb/pull/15619)
```
Comment (TomShawn) [inline releases/3.1.0-rc.md:21]:
```suggestion
    - 支持 `YEAR`、`MONTH`、`TO_DAY` 函数作为 Hash partition table 的分区 key [#15619](https://github.com/pingcap/tidb/pull/15619)
```

File: releases/3.1.0-rc.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+    - KV server 支持 TLS[#7305](https://github.com/tikv/tikv/pull/7305)
+
++ PD
+    - `shuffle-region-scheduler` 支持调度 learner [#2235](https://github.com/pingcap/pd/pull/2235)
+    - pd-ctl 增加配置 Placement Rules 的命令 [#2306](https://github.com/pingcap/pd/pull/2306)
+
++ Tools
+    - TiDB Binlog
+        * 同步链路新增 TLS 功能 [#931](https://github.com/pingcap/tidb-binlog/pull/931)[#937](https://github.com/pingcap/tidb-binlog/pull/937)[#939](https://github.com/pingcap/tidb-binlog/pull/939)
+        * Drainer 新增 `kafka-client-id` 配置项，支持连接 Kafka 客户端配置客户端 ID[#929](https://github.com/pingcap/tidb-binlog/pull/929)
```
Comment (TomShawn) [inline releases/3.1.0-rc.md:42]:
```suggestion
        * Drainer 新增 `kafka-client-id` 配置项，支持连接 Kafka 客户端配置客户端 ID [#929](https://github.com/pingcap/tidb-binlog/pull/929)
```

File: releases/3.1.0-rc.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+    - TiDB Lightning
+        * 优化 Lightning 的性能 [#281](https://github.com/pingcap/tidb-lightning/pull/281)[#275](https://github.com/pingcap/tidb-lightning/pull/275)
```
Comment (TomShawn) [inline releases/3.1.0-rc.md:44]:
```suggestion
        * 优化 Lightning 的性能 [#281](https://github.com/pingcap/tidb-lightning/pull/281) [#275](https://github.com/pingcap/tidb-lightning/pull/275)
```

File: releases/3.1.0-rc.md [hunk 8]
```diff
@@ -0,0 +1,92 @@
+---
+title: TiDB 3.1.0 RC Release Notes
```
Comment (TomShawn) [inline releases/3.1.0-rc.md:2]:
```suggestion
title: TiDB 3.1 RC Release Notes
```

File: releases/3.1.0-rc.md [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+category: Releases
+---
+
+# TiDB 3.1.0 RC Release Notes
```
Comment (TomShawn) [inline releases/3.1.0-rc.md:6]:
```suggestion
# TiDB 3.1 RC Release Notes
```

**Comment by TomShawn** [comment]
LGTM

**Comment by TomShawn** [comment]
@scsldb PTAL, thanks!

**Comment by scsldb** [comment]
LGTM

**Pull request merged**